libs/framework/src/bundle_context.c

```diff
@@ -1208,9 +1208,9 @@ bool celix_bundleContext_useServiceWithOptions(
     long eventId = celix_framework_fireGenericEvent(ctx->framework, -1, celix_bundle_getId(ctx->bundle), "create service tracker for celix_bundleContext_useServiceWithOptions", &data, celix_bundleContext_useServiceWithOptions_1_CreateServiceTracker, NULL, NULL);
     celix_framework_waitForGenericEvent(ctx->framework, eventId);
 
-    celix_framework_waitForEmptyEventQueue(ctx->framework); //ensure that a useService wait if a listener hooks concept, which triggers an async service registration
```

Comment:
This is not necessary for the indirect call path, since the following generic event sync with these async service registration.